src/gui/qgsfilecontentsourcelineedit.cpp

```diff
@@ -269,97 +269,101 @@ QString QgsAbstractFileContentSourceLineEdit::settingsKey() const
   return mLastPathKey.isEmpty() ? defaultSettingsKey() : mLastPathKey;
 }
 
-//
-// QgsSvgSourceLineEdit
-//
-
-///@cond PRIVATE
-
-QString QgsSvgSourceLineEdit::fileFilter() const
-{
-  return tr( "SVG files" ) + " (*.svg)";
-}
-
-QString QgsSvgSourceLineEdit::selectFileTitle() const
-{
-  return tr( "Select SVG File" );
-}
-
-QString QgsSvgSourceLineEdit::fileFromUrlTitle() const
-{
-  return tr( "SVG From URL" );
-}
-
-QString QgsSvgSourceLineEdit::fileFromUrlText() const
+void QgsAbstractFileContentSourceLineEdit::setMessageBar( QgsMessageBar *bar )
 {
-  return tr( "Enter SVG URL" );
+  mMessageBar = bar;
 }
 
-QString QgsSvgSourceLineEdit::embedFileTitle() const
+QgsMessageBar *QgsAbstractFileContentSourceLineEdit::messageBar() const
 {
-  return tr( "Embed SVG File" );
+  return mMessageBar;
 }
 
-QString QgsSvgSourceLineEdit::extractFileTitle() const
-{
-  return tr( "Extract SVG File" );
-}
 
-QString QgsSvgSourceLineEdit::defaultSettingsKey() const
-{
-  return QStringLiteral( "/UI/lastSVGDir" );
-}
-///@endcond
 
 //
-// QgsImageSourceLineEdit
+// QgsPictureSourceLineEditBase
 //
 
 ///@cond PRIVATE
 
-QString QgsImageSourceLineEdit::fileFilter() const
+QString QgsPictureSourceLineEditBase::fileFilter() const
 {
-  return tr( "All files" ) + " (*.*)";
+  switch ( mFormat )
+  {
+    case Svg:
+      return tr( "SVG files" ) + " (*.svg)";
+    case Image:
+      return tr( "All files" ) + " (*.*)";
```

Comment:
You could use QImageReader::supportedImageFormats() here to narrow down list to only supported image formats.